### PR TITLE
chore: store evaluation count in redis

### PR DESCRIPTION
**Pull request by ono-max**

### CODE REVIEW

File: pkg/eventpersister/persister/persister.go [hunk 1]
```diff
@@ -645,3 +662,25 @@ func (p *Persister) createUserEvent(
...
+		uck := p.key(userCountKey, e.FeatureId, e.VariationId, e.Timestamp)
+		_, err = p.evaluationCountCacher.PFAdd(uck, []string{e.UserId})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *Persister) key(kind, featureID, variationID string, timestamp int64) string {
```
Comment (cre8ivejp) [inline pkg/eventpersister/persister/persister.go:682]: Sorry for mentioning this now, but we also need the `environmentNamespace` in the key.

- Event count: `ec:environment_namespace:feature_flag_id:variation_id:daily_timestamp`
- User count: `uc:environment_namespace:feature_flag_id:variation_id:daily_timestamp`

File: pkg/cache/cache.go [hunk 2]
```diff
@@ -59,6 +60,11 @@ type Deleter interface {
 	Delete(key string) error
 }
 
+type Counter interface {
+	Increment(key string) (int64, error)
+	PFAdd(key string, els []string) (int64, error)
```
Comment (cre8ivejp) [inline pkg/cache/cache.go:65]: We should change the interface to:

E.g. `PFAdd(key string, els ...string) (int64, error)`

Also in here:
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/redis/v3/redis.go#L66

Reference:
https://pkg.go.dev/github.com/fiorix/go-redis/redis#Client.PFAdd

**Review by cre8ivejp — CHANGES_REQUESTED** [review]

**Comment by ono-max** [comment]
@cre8ivejp 

I've updated 👍

**Review by masaaania — APPROVED** [review]
LGTM!

**Review by cre8ivejp — APPROVED** [review]
Nice!
Thanks!

**Pull request merged**